compiler/il/OMRNode.cpp

```diff
@@ -3083,28 +3087,77 @@ OMR::Node::getMonitorClass(TR_ResolvedMethod * vmMethod)
    return 0;
    }
 
+void *
+OMR::Node::getMonitorInfoInNode()
+   {
+   TR_ASSERT(self()->hasMonitorInfoInNode(),"Node %p does not have monitor info in Node");
+   TR::Node  * info = (self()->getOpCodeValue() != TR::tstart) ? self()->getChild(1) : self()->getChild(4);
+   return (void *) info;
+   }
+
+void
+OMR::Node::setMonitorInfoInNode(void * info)
+   {
+   TR_ASSERT(self()->getOpCodeValue() == TR::tstart || self()->getOpCodeValue() == TR::monent || self()->getOpCodeValue() == TR::monexit,
```

Comment:
Shouldn't this be assert fatal since this is a new API?

Reply:
I am thinking even if the node is not the expected type the execution might still be able to finish.